### PR TITLE
Implement population generation

### PR DESCRIPTION
**Pull request by lynn**
Whew, this should satisfy all the requirements in the spec…

The `Atlas` field I added to the `Population` class is essentially a map from (cluster ID, cluster type) pairs to `GeoPosition`s. This information is irrelevant for the simulator, but imporant for the visualizer.

The massive `Generate` method is split into “regions”. I think I see a way to split it up into smaller methods, but I won’t bother with it right now.

### CODE REVIEW

File: src/main/cpp/core/Atlas.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+{
+public:
+	using Key = std::pair<std::size_t, ClusterType>;
+	using Map = std::map<Key, geo::GeoPosition>;
+
+	Atlas() {}
+	Atlas(const Map& map) : map(map) {}
+
+	/// Look up a cluster's position. If it isn't found, throw std::out_of_range.
+	const geo::GeoPosition& Lookup(const Key& key) { return map.at(key); }
```
Comment (jonathanvdc) [inline src/main/cpp/core/Atlas.h]: Can you make this method `const`?

File: src/main/cpp/pop/Generator.cpp [hunk 2]
```diff
@@ -5,187 +5,307 @@
...
+	using WorkClusterID = ClusterID;
+	using CommunityClusterID = ClusterID;
+	using HouseholdClusterID = ClusterID;
+	using School = std::vector<SchoolClusterID>;
+	using College = School;
+
+	// Logging.
+	auto console = spdlog::stderr_logger_st("popgen");
+	console->set_level(spdlog::level::debug);
+	console->set_pattern("\x1b[36;1m[popgen] %v\x1b[0m");
```
Comment (jonathanvdc) [inline src/main/cpp/pop/Generator.cpp:46]: :heart: ANSI escape codes.

File: src/main/cpp/core/Cluster.h [hunk 3]
```diff
@@ -36,17 +36,16 @@ namespace stride {
 class RngHandler;
 class Calendar;
 
+using ClusterID = std::size_t;
```
Comment (jonathanvdc) [inline src/main/cpp/core/Cluster.h]: Maybe you should rename this to `ClusterId` for consistency with `PersonId` and `RegionId`. I've noticed that `CityID` and `ProvinceID` also use the all--upper-case suffix. Alternatively, you can rename `PersonId` and `RegionId` to match your `*ID` naming convention.

File: src/main/cpp/geo/Profile.h [hunk 4]
```diff
@@ -14,15 +14,27 @@ using ProfileRef = std::shared_ptr<const Profile>;
...
+	}
 
 	/// Return a list of cities in this geodistribution profile.
-	const std::vector<City>& GetCities() const { return cities; }
+	/// The result is sorted in descending order of relative population.
+	const std::vector<City>& GetCities() const { return m_cities; }
+
+	/// Return the simulation area, which is defined as the the latitude/longitude
+	/// bounding geodesic rectangle of all the cities.
+	const GeoRectangle GetSimulationArea() const { return m_simulation_area; }
```
Comment (jonathanvdc) [inline src/main/cpp/geo/Profile.h]: Returning a `const` value doesn't actually mean anything. Please delete the leading `const` or change the return type to a reference.

**Review by jonathanvdc — APPROVED** [review]
LGTM. I've made some minor remarks that you might want to review.

**Comment by lynn** [comment]
Performed changes as requested; also deleted some dead commented-out code from `Generator.h`. Should be good to go!

**Pull request merged**